grpc/errors.go

```diff
@@ -38,13 +39,21 @@ func wrapError(ctx context.Context, err error) error {
 			jsonSubErrs, err := json.Marshal(berr.SubErrors)
 			if err != nil {
 				return berrors.InternalServerError(
+					0,
 					"error marshaling json SubErrors, orig error %q",
 					err)
 			}
 			pairs = append(pairs, "suberrors")
 			pairs = append(pairs, string(jsonSubErrs))
 		}
 
+		// If there is a RetryAfter value then extend the metadata pairs to
+		// include the value.
+		if berr.RetryAfter > 0 {
```

Comment:
time.Duration can be negative. Although having a negative retryAfter would be _very_ weird, I'd rather have that faithfully carried through rather than silently dropped.
```suggestion
		if berr.RetryAfter != 0 {
```